core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorTableMetadata.java

```diff
@@ -72,13 +81,19 @@ public Optional<String> getComment()
         return comment;
     }
 
+    public List<String> getCheckConstraints()
```

Comment:
Add javadoc explaining how these String values should be understood, i.e. what can and what cannot be a check constraint value.

I guess the intention here is "String representation of a Trino SQL scalar expression that can refer to table columns by name and produces a result coercible to boolean"

Reply:
The SQL spec already has a notion of constraints, so these need to be modeled in a way consistent to what the spec describes. Have you looked at that, yet? By the way, the spec refers to various forms of constraints as "integrity constraints".

Reply:
@ebyhr what about splitting this PR into  SPI & Engine side and the rest? Or maybe even SPI part, then engine, then the rest?
We need Martin's attention especially for the SPI part.

Reply:
Sure, I will separate this PR.

Reply:
Sent #14964 for engine and SPI.